spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/RoutingKeyGenerator.scala

```diff
@@ -44,8 +44,12 @@ class RoutingKeyGenerator(table: TableDef, columnNames: Seq[String])
 
   private def fillRoutingKey(stmt: BoundStatement): Array[ByteBuffer] = {
     val rk = routingKey.get
-    for (i <- 0 until partitionKeyIdxs.size)
+    for (i ← partitionKeyIdxs.indices) {
+      if (stmt.isNull(partitionKeyIdxs(i)))
+        throw new NullPointerException(
```

Comment:
What about:

         throw new NullOrEmptyColumnException(s"Invalid null value for partition key part $theValue")

         @SerialVersionUID(1L)
          class NullOrEmptyColumnException(columnName: String, message: String, cause: Throwable) extends             RuntimeException(message, cause) {

               def this(columnName: String) = this(columnName,"", null) 
           }